src/sentry/api/endpoints/organization_member/team_details.py

```diff
@@ -44,27 +46,41 @@ def serialize(
         }
 
 
-class RelaxedOrganizationPermission(OrganizationPermission):
-    _allowed_scopes = [
-        "org:read",
-        "org:write",
-        "org:admin",
-        "member:read",
-        "member:write",
-        "member:admin",
-    ]
+class TeamOrgMemberPermission(OrganizationPermission):
 
     scope_map = {
-        "GET": _allowed_scopes,
-        "POST": _allowed_scopes,
-        "PUT": _allowed_scopes,
-        "DELETE": _allowed_scopes,
+        "GET": [
+            "org:read",
+            "org:write",
+            "member:read",
+            "member:write",
+            "member:admin",
+        ],
+        "POST": ["org:read", "org:write", "team:write"],
+        "PUT": [
+            "org:read",
+            "org:write",
+            "member:read",
+            "member:write",
+            "member:admin",
+        ],
+        "DELETE": ["org:read", "org:write", "team:write"],
     }
 
 
+def _has_elevated_scope(access: Access) -> bool:
+    # validate the token has more than just org:read
+    return access.has_scope("org:write") or access.has_scope("team:write")
+
+
+def _is_org_owner_or_manager(access: Access) -> bool:
+    roles = access.get_organization_roles()
+    return "owner" in roles or "manager" in roles
```

Comment:
Is there a way to pull from some org role enum instead of hard coding these as strings?